pkg/controller/periodic/job.go

```diff
@@ -124,6 +124,10 @@ func (j *JobController) WaitForJobCompletion(ctx context.Context, jobName string
 				return fmt.Errorf("watcher channel was closed unexpectedly")
 			}
 
+			if event.Type == watch.Deleted {
+				return nil
+			}
```

Comment:
This is also pretty import, because one can delete the job and in that case it would end up in infinite loop (never ending watch)